gunicorn/reloader.py

```diff
@@ -54,17 +54,18 @@ def run(self):
             time.sleep(self._interval)
 
 
-has_inotify = False
-if sys.platform.startswith('linux'):
-    try:
-        from inotify.adapters import Inotify
-        import inotify.constants
-        has_inotify = True
-    except ImportError:
-        pass
+class OSNoInotifyError(ImportError):
+    pass
+
 
+has_inotify = False
+try:
+    if not sys.platform.startswith('linux'):
+        raise OSNoInotifyError("The inotify mechanism is only supported on Linux")
```

Comment:
I think this was maybe nicer when the `try/except` was tightly scoped to just the import.